src/redisraft.c

```diff
@@ -597,6 +621,11 @@ static bool handleInterceptedCommands(RedisRaftCtx *rr,
         return true;
     }
 
+    if (len == strlen("UNWATCH") && strncasecmp(cmd_str, "UNWATCH", len) == 0) {
```

Comment:
also `WATCH` can be inside of multi-exec?. We need to check Redis' behavior and mimic it here.

Reply:
WATCH can't be inside a multiexec (tested previously).  Apparently, UNWATCH can be.  However, it seems to always cause the command to return nil.  the redis semantics seem to be poor here.

Reply:
change that.  nil on dirty is what exec always returns.  so unwatch within a multi/exec dirties the session.  weird.  In this case, I'd think we would have to think to decide what to do, as we aren't (currently) queing the multi/exec on the log, we can know at unwatch time if we are in an multi and just error it out.

Reply:
so an even more update.  unwatch is usable in multi (so current code is problematic), BUT, it doesn't do anything in reality.  i.e. if the watched value in concurrently modified, nil is returned (as multi is dirty), which makes sense as UNWATCH only would execute on exec, but already dirty).  And if its not dirty, you just get an OK back.  It would make more sense to reject it, like it rejects WATCH

(error) ERR WATCH inside MULTI is not allowed

Reply:
refactored, so UNWATCH will be handled in normal redis command path so will be enqueued into a MULTI (pointless, but what redis does)

Reply:
Okay, are we going to prevent WATCH inside multi in this PR or in another PR? 
Btw, I guess we can even just return OK to UNWATCH without adding it to the queue.

Reply:
prevent watch within MULTI.

Reply:
pff, horrible. I hope we can reject unwatch as well